Bio/PDB/Residue.py

```diff
@@ -115,6 +119,63 @@ def get_atoms(self):
         """Return atoms."""
         yield from self
 
+    def mutate(self, selected_rotamer):
+        """
+        Mutates a given residue based on the best fitting rotamers
+        :param mutate_to: Three letter amino acid code in uppercase letters
+        :param sample_residue: Sample residue generated from a sampler RotamerSampling
+        :param selected_rotamer: Selected rotamer generate from a sampler from RotamerSampling
+        """
+        import Bio.PDB.Atom  # Moved this here to avoid circular import
+
+        mutate_to = selected_rotamer.resname
+        sample_residue = selected_rotamer.get_sample_residue()
+        if mutate_to not in ["ALA", "GLY"]:
+            # Introduce the rotamer
+            for angle in ["CHI1", "CHI2", "CHI3", "CHI4"]:
+                if mutate_to not in CHI_ANGLES[angle]:
+                    continue
+                dihedral_start = calc_dihedral(
+                    *[
+                        Vector(sample_residue[x])
+                        for x in CHI_ANGLES[angle][mutate_to]["ref_plane"]
+                    ]
+                )
+                rotation_angle = dihedral_start - np.deg2rad(selected_rotamer[angle])
+                axis = CHI_ANGLES[angle][mutate_to]["axis"]
+                # print(angle)
+                for atom in RESIDUE_ORDER[mutate_to][
+                    RESIDUE_ORDER[mutate_to].index(axis[1]) + 1 :
+                ]:
+                    sample_residue[atom] = (
+                        np.dot(
+                            rotaxis2m(
+                                rotation_angle,
+                                Vector(
+                                    sample_residue[axis[0]] - sample_residue[axis[1]]
+                                ),
+                            ),
+                            sample_residue[atom] - sample_residue[axis[1]],
+                        )
+                        + sample_residue[axis[1]]
+                    )
+        for atom, coord in sample_residue.items():
+            if atom not in {"C", "N", "CA", "O"}:
+                new_atom = Bio.PDB.Atom.Atom(
+                    name=atom,
+                    element=atom[0],
+                    fullname="{}{}".format(
+                        " " * (4 - len(atom)), atom
+                    ),  # for writing the structure, should be 4-char long
+                    coord=np.asarray(coord),
+                    bfactor=1.0,
+                    altloc=" ",
+                    occupancy=1.0,
+                    serial_number=9999,  # does not matter much, only for writing the struct.
```

Comment:
Yes it would be more work and convolution, but it would be way cooler if the serial numbers were preserved for only rotamer changes.  For changes where the residue is actually mutated it seems like the serial numbers could be copied over using the default atom ordering within a residue - this is lazier than matching the ATOM letter codes and might be the same in most cases.  For new atoms from an actual mutation to another residue new numbers would need to be generated, possibly with a counter to distinguish them (the field is actually 5 digits, so up to 99,999).

I think the default when writing PDB files is to regenerate the serial numbers unless `preserve_atom_numbering` is set, so arguably this gets wiped out anyway - but it does seem like a user might want to track atoms between alternate structures/configurations when using this code.